types/aws-lambda/trigger/cloudformation-custom-resource.d.ts

```diff
@@ -1,70 +1,85 @@
 import { Handler } from "../handler";
 
+export interface ResourcePropertiesCommon extends Record<string, any> {
+    ServiceToken: string;
+    ServiceTimeout?: string;
+}
+
 // Note, responses are *not* lambda results, they are sent to the event ResponseURL.
-export type CloudFormationCustomResourceHandler = Handler<CloudFormationCustomResourceEvent, void>;
+export type CloudFormationCustomResourceHandler<
+    TResourceProperties extends ResourcePropertiesCommon = ResourcePropertiesCommon,
```

Comment:
I think these might be able to be simplified by not requiring the bounds and using an intersection type at the final use, but it's a tradeoff, so 🤷

Eg, below, something like:

```
  ResourceProperties: TResourceProperties & ResoucePropertiesCommon;
```

Reply:
Thanks for this suggestion! After some testing in my own codebase I found I actually really prefer doing it this way with the type union - not only does it make the typings more readable but it was a better solution to using these typings for CDK custom resources where you don't want to specify the `ServiceToken` and `ServiceTimeout` since the provider framework does that for you